Apps/Src/Temperature.c

```diff
@@ -420,22 +419,42 @@ int32_t Temperature_GetMaxTemperature(void) {
  * @brief Gives fan speed based on Average temperature of pack and past error values
  * @param InputTemp - current temperature
  * @param DesiredTemp - desired temperature
- * @return FanSpeed: 0-4000 PWM
+ * @return FanSpeed: 0-8
  */
 int32_t Temperature_PID_Output(int32_t InputTemp, int32_t DesiredTemp) {
     Error = DesiredTemp - InputTemp;
-    ErrorSum = ErrorSum + Error;
-
-    if (PreviousError == 0) {PreviousError = Error;} //init previous val first time
-
-    Rate = Error - PreviousError;
-    PreviousError = Error;     //updates previous err value
-
-    if (((PROPORTION*(Error) + INTEGRAL*(ErrorSum) + DERIVATIVE*(Rate))/DIVISOR) > MAX_FAN_SPEED) {
-        return MAX_FAN_SPEED;
-    }
-    if (((PROPORTION*(Error) + INTEGRAL*(ErrorSum) + DERIVATIVE*(Rate))/DIVISOR) <= 0) {
-        return 0;
-    }
-    return (PROPORTION*(Error) + INTEGRAL*(ErrorSum) + DERIVATIVE*(Rate))/DIVISOR;
+	
+	//Only read error sum in range
+	if(abs(Error) < I_ZONE){
+		ErrorSum += Error;
+	}else{
+		ErrorSum = 0;
+	}
+
+	//Cap error sum at 500 read degrees
+	if(abs(ErrorSum) > I_MAX_ACCUM){
+		if(ErrorSum > 0){
```

Comment:
just do ErrorSum = abs(I_MAX_ACCUM)?

Reply:
I don't think that would work because you're supposed to see if error goes over the max accum

Reply:
nah i did that because error sum needs to be able to be negative

Reply:
could replace all these if elses with ternaries

Reply:
fixed

Reply:
fixed